bioagents/msa/msa.py

```diff
@@ -936,7 +938,7 @@ def _regularize_input(self, *entities, **params):
     def _iter_stmts(self, stmts):
         for stmt in stmts:
             ags = stmt.agent_list()
-            if self._role == 'OTHER':
+            if self._role == 'OTHER' or isinstance(stmt, Complex):
```

Comment:
This change has been sitting in my bioagents for a while, and I am afraid I don't entirely recall the motivation. It does not fail any tests but it would be worth taking a look to make sure it improves the behavior of the bioagent.

Reply:
I think this makes sense. As far as I can tell `self._role` is never `OTHER` because there isn't an instance of the `_Commons` class that uses that `_role` (it was maybe something we planned to support but didn't need it in the end). Adding the `isinstance(stmt, Complex)` ensures that we don't just pick `ags[0]` or `ags[1]` from a Complex statement that has arbitrary agent ordering when searching for up/downstreams. Since all the Complex agents (including the one we queries) is included in the yield below, we need to then exclude it later, as is done in the other change  on line 988.